Starwatch.Core/API/Rest/Route/AnnouncementRoute.cs

```diff
@@ -0,0 +1,387 @@
+                    }
+                    catch (Exception ex)
+                    {
+                        mon.Unlock();
+                        return new RestResponse(RestStatus.InternalError, "Failed to save config: " + ex.Message);
+                    }
+                }
+
+                mon.Unlock();
+                return new RestResponse(RestStatus.ResourceNotFound, MISSING_RESOURCE);
+            }
+            else
+            {
+                return new RestResponse(RestStatus.BadRequest, "id must be included.");
+            }
+        }
+
+#if DEBUG
+        private AnnouncementPatch FakeBody (Query query)
+        {
+            if (!Debugger.IsAttached)
+            {
+                // shouldn't get here.
+                throw new Exception("Phishing attempt.");
+            }
+
+            AnnouncementPatch ap = new AnnouncementPatch
+            {
+                Message = "",
+                Enabled = false,
+                Interval = 1800
+            };
+
+            if (query.ContainsKey("message"))
+            {
+                ap.Message = query["message"];
+            }
+
+            if (query.ContainsKey("enabled"))
+            {
+                bool e = false;
+                if (bool.TryParse(query["enabled"], out e))
+                {
+                    ap.Enabled = e;
+                }
+            }
+
+            if (query.ContainsKey("interval"))
+            {
+                double interval2 = 0d;
+                if (double.TryParse(query["interval"], out interval2))
+                {
+                    ap.Interval = interval2;
+                }
+            }
+
+            return ap;
+        }
+#endif
+
+        /// <summary>
+        /// Updates the announcement
+        /// </summary>
+        /// <param name="query"></param>
+        /// <param name="payloadObject"></param>
+        /// <returns></returns>
+        public override RestResponse OnPatch(Query query, object payloadObject)
+        {
+#if DEBUG
+            if (Debugger.IsAttached)
+            {
+                if (payloadObject == null)
+                {
+                    payloadObject = FakeBody(query);
+                }
+            }
+#endif
+
+            if (payloadObject == null)
+            {
+                return new RestResponse(RestStatus.BadRequest, "No payload.");
+            }
+
+            AnnouncementPatch payload = (AnnouncementPatch) payloadObject;
+
+            if (payload.Id < 0)
+            {
+                return new RestResponse(RestStatus.BadRequest, ID_MUST_BE_POSITIVE);
+            }
+
+            // Grab the configurator from the monitor 
+            AnnouncementMonitor mon = null;
+            bool hasMon = Starbound.TryGetMonitor(out mon);
+
+            if (!hasMon)
+            {
+                return new RestResponse(RestStatus.InternalError, NO_MONITOR);
+            }
+
+            if (!mon.Lock())
```

Comment:
I dont know what these locks are doing. Is this some kind of race-condition check?